### ISSUE TITLE
false positive @typescript-eslint/no-unsafe-assignment when using function bindings

### ISSUE THREAD

**Issue by Theo-Steiner** [ota-meshi/svelte-eslint-parser#741]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I'm using [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte). (`*.svelte` file linting does not work with the parser alone. You should also use eslint-plugin-svelte with it.)
- [x] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte) repo and open the issue in [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte) repo if there is no solution.
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.33.0

### What version of `eslint-plugin-svelte` and `svelte-eslint-parser` are you using?

- eslint-plugin-svelte@3.11.0
- svelte-eslint-parser@1.3.1

### What did you do?

<details>
<summary>configuration</summary>
import js from "@eslint/js";
import sveltePlugin from "eslint-plugin-svelte";
import tseslint from "typescript-eslint";
export default tseslint.config(
  js.configs.recommended,
  ...tseslint.configs.recommended,
  ...sveltePlugin.configs["flat/recommended"],
  {
    files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
    languageOptions: {
      parserOptions: {
        parser: tseslint.parser,
        extraFileExtensions: [".svelte"],
        projectService: true
      },
    }
  },
  {
  	rules: {
			'@typescript-eslint/no-unsafe-assignment': 'error',
		}
  }
);
</details>

```svelte
<script lang="ts">
    let a = 'hello';
</script>

<input bind:value={
    () => a,
    (v) => a = v
}>
```

### What did you expect to happen?

when using [function bindings](https://svelte.dev/docs/svelte/bind#Function-bindings), typescript  v seems to be inferred correctly as a string by typescript, so I assumed `a = v` would be a safe assignment. (`string` to `string`)

### What actually happened?

for some reason, v is parsed as `any` however, triggering the rule `@typescript-eslint/no-unsafe-assignment`

### Link to **GitHub Repo** with Minimal Reproducible Example

[playground](https://eslint-online-playground.netlify.app/#eNqNU01vnDAQ/SuWLySrBe4ku4rUj1vVSj3GObgwUFNjI9ugjSL+e8cfsJtN+gEHNDPvPc+MHy/Umrr88vxBD6NWoFxhZ5AOaEXvbW3E6Ijkqjsw6iyjR6YIPhIc4eRAsp8gpc7umLovIxgBGAg1To78EKqpZi4nOLxE3s0tORwJ36dojiEKzUwtR7rHM2qtWtEVvdUKWwg8RoWq5dQAoxV5jFwW2t6lZhlNkpgOiaIpfLs++YTSqAxWChxuk0dxgRMbR3pLWqMH5D5EUNkjFWdK9aj4TU6dUCsyAvMxJPO1iTPH2YhY8e55hLigPBYCGE4B3EDLJ3kmpSZvfPe9TZEtDNR6GEA10IRpiwJnfMV4F3PZ/op7ZLSV3JUXcEafAiPdVCskWNx2ttttW8725DLE068zvc2iCloETTPxDr6OTmiFUkmYkJEbC+Ztfq1U50XERFL0D5yc4Z+xt08nB8pGBZxm88F6fJAzuofafQczixpQ1kywVpeEW/wnBtFrCAqTY8Scf7OHN5dXKp1PyvIWcm6t6BSu0GUVycAYbTJU88QozdTtHdpv5PUv3MaVr72yNzWjg27w3ORjRhuYP8LoL0bVAtCO255eu9RT8SLBoqG2X2B12F+K1859F/ofxbSQPF7UH6Dn/f0TsP0dl7i0x4UuvwFdl475)
